### ISSUE TITLE
how can i solve this problem(i just start learning)?, really appreciate it , thank you~

### ISSUE THREAD

**Issue by murrayliu** [BVLC/caffe#6197]
I0125 15:54:25.500926 13112 caffe.cpp:218] Using GPUs 0
I0125 15:54:25.530297 13112 caffe.cpp:223] GPU 0: GeForce GTX 1060 6GB
I0125 15:54:25.761474 13112 solver.cpp:44] Initializing solver from parameters:
test_iter: 1000
test_interval: 1000
base_lr: 0.01
display: 20
max_iter: 450000
lr_policy: "step"
gamma: 0.1
momentum: 0.9
weight_decay: 0.0005
stepsize: 100000
snapshot: 10000
snapshot_prefix: "models/bvlc_reference_caffenet/caffenet_train"
solver_mode: GPU
device_id: 0
net: "models/bvlc_reference_caffenet/train_val.prototxt"
train_state {
  level: 0
  stage: ""
}
I0125 15:54:25.761608 13112 solver.cpp:87] Creating training net from net file: models/bvlc_reference_caffenet/train_val.prototxt
I0125 15:54:25.761895 13112 net.cpp:294] The NetState phase (0) differed from the phase (1) specified by a rule in layer data
I0125 15:54:25.761912 13112 net.cpp:294] The NetState phase (0) differed from the phase (1) specified by a rule in layer accuracy
I0125 15:54:25.762094 13112 net.cpp:51] Initializing net from parameters:
name: "CaffeNet"
state {
  phase: TRAIN
  level: 0
  stage: ""
}
layer {
  name: "data"
  type: "Data"
  top: "data"
  top: "label"
  include {
    phase: TRAIN
  }
  transform_param {
    mirror: true
    crop_size: 227
    mean_file: "data/ilsvrc12/imagenet_mean.binaryproto"
  }
  data_param {
    source: "examples/imagenet/ilsvrc12_train_lmdb"
    batch_size: 256
    backend: LMDB
  }
}
layer {
  name: "conv1"
  type: "Convolution"
  bottom: "data"
  top: "conv1"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 96
    kernel_size: 11
    stride: 4
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0
    }
  }
}
layer {
  name: "relu1"
  type: "ReLU"
  bottom: "conv1"
  top: "conv1"
}
layer {
  name: "pool1"
  type: "Pooling"
  bottom: "conv1"
  top: "pool1"
  pooling_param {
    pool: MAX
    kernel_size: 3
    stride: 2
  }
}
layer {
  name: "norm1"
  type: "LRN"
  bottom: "pool1"
  top: "norm1"
  lrn_param {
    local_size: 5
    alpha: 0.0001
    beta: 0.75
  }
}
layer {
  name: "conv2"
  type: "Convolution"
  bottom: "norm1"
  top: "conv2"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 256
    pad: 2
    kernel_size: 5
    group: 2
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 1
    }
  }
}
layer {
  name: "relu2"
  type: "ReLU"
  bottom: "conv2"
  top: "conv2"
}
layer {
  name: "pool2"
  type: "Pooling"
  bottom: "conv2"
  top: "pool2"
  pooling_param {
    pool: MAX
    kernel_size: 3
    stride: 2
  }
}
layer {
  name: "norm2"
  type: "LRN"
  bottom: "pool2"
  top: "norm2"
  lrn_param {
    local_size: 5
    alpha: 0.0001
    beta: 0.75
  }
}
layer {
  name: "conv3"
  type: "Convolution"
  bottom: "norm2"
  top: "conv3"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 384
    pad: 1
    kernel_size: 3
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0
    }
  }
}
layer {
  name: "relu3"
  type: "ReLU"
  bottom: "conv3"
  top: "conv3"
}
layer {
  name: "conv4"
  type: "Convolution"
  bottom: "conv3"
  top: "conv4"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 384
    pad: 1
    kernel_size: 3
    group: 2
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 1
    }
  }
}
layer {
  name: "relu4"
  type: "ReLU"
  bottom: "conv4"
  top: "conv4"
}
layer {
  name: "conv5"
  type: "Convolution"
  bottom: "conv4"
  top: "conv5"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 256
    pad: 1
    kernel_size: 3
    group: 2
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 1
    }
  }
}
layer {
  name: "relu5"
  type: "ReLU"
  bottom: "conv5"
  top: "conv5"
}
layer {
  name: "pool5"
  type: "Pooling"
  bottom: "conv5"
  top: "pool5"
  pooling_param {
    pool: MAX
    kernel_size: 3
    stride: 2
  }
}
layer {
  name: "fc6"
  type: "InnerProduct"
  bottom: "pool5"
  top: "fc6"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  inner_product_param {
    num_output: 4096
    weight_filler {
      type: "gaussian"
      std: 0.005
    }
    bias_filler {
      type: "constant"
      value: 1
    }
  }
}
layer {
  name: "relu6"
  type: "ReLU"
  bottom: "fc6"
  top: "fc6"
}
layer {
  name: "drop6"
  type: "Dropout"
  bottom: "fc6"
  top: "fc6"
  dropout_param {
    dropout_ratio: 0.5
  }
}
layer {
  name: "fc7"
  type: "InnerProduct"
  bottom: "fc6"
  top: "fc7"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  inner_product_param {
    num_output: 4096
    weight_filler {
      type: "gaussian"
      std: 0.005
    }
    bias_filler {
      type: "constant"
      value: 1
    }
  }
}
layer {
  name: "relu7"
  type: "ReLU"
  bottom: "fc7"
  top: "fc7"
}
layer {
  name: "drop7"
  type: "Dropout"
  bottom: "fc7"
  top: "fc7"
  dropout_param {
    dropout_ratio: 0.5
  }
}
layer {
  name: "fc8"
  type: "InnerProduct"
  bottom: "fc7"
  top: "fc8"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  inner_product_param {
    num_output: 1000
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0
    }
  }
}
layer {
  name: "loss"
  type: "SoftmaxWithLoss"
  bottom: "fc8"
  bottom: "label"
  top: "loss"
}
I0125 15:54:25.762370 13112 layer_factory.hpp:77] Creating layer data
I0125 15:54:25.762460 13112 db_lmdb.cpp:35] Opened lmdb examples/imagenet/ilsvrc12_train_lmdb
I0125 15:54:25.762480 13112 net.cpp:84] Creating Layer data
I0125 15:54:25.762487 13112 net.cpp:380] data -> data
I0125 15:54:25.762506 13112 net.cpp:380] data -> label
I0125 15:54:25.762518 13112 data_transformer.cpp:25] Loading mean file from: data/ilsvrc12/imagenet_mean.binaryproto
F0125 15:54:25.763913 13112 blob.cpp:133] Check failed: data_
*** Check failure stack trace: ***
    @     0x7ff4f09275cd  google::LogMessage::Fail()
    @     0x7ff4f0929433  google::LogMessage::SendToLog()
    @     0x7ff4f092715b  google::LogMessage::Flush()
    @     0x7ff4f0929e1e  google::LogMessageFatal::~LogMessageFatal()
    @     0x7ff4f0f30e3b  caffe::Blob<>::mutable_cpu_data()
    @     0x7ff4f0f3dda6  caffe::Blob<>::FromProto()
    @     0x7ff4f0fa037b  caffe::DataTransformer<>::DataTransformer()
    @     0x7ff4f0ff3f29  caffe::BaseDataLayer<>::LayerSetUp()
    @     0x7ff4f0ff404e  caffe::BasePrefetchingDataLayer<>::LayerSetUp()
    @     0x7ff4f10ccc97  caffe::Net<>::Init()
    @     0x7ff4f10cf3de  caffe::Net<>::Net()
    @     0x7ff4f0fa5e55  caffe::Solver<>::InitTrainNet()
    @     0x7ff4f0fa7245  caffe::Solver<>::Init()
    @     0x7ff4f0fa755f  caffe::Solver<>::Solver()
    @     0x7ff4f0f592e1  caffe::Creator_SGDSolver<>()
    @           0x40aa08  train()
    @           0x407300  main
    @     0x7ff4ef897830  __libc_start_main
    @           0x407b29  _start
    @              (nil)  (unknown)
Aborted (core dumped)

i guess the issue is a easy one and may cause by the setting of some module, since i always get the same error even use different net, like alexnet and googlenet~

**Comment by Noiredd** [comment]
This is most likely caused by the mean file being malformed or non existent. Check the path `data/ilsvrc12/imagenet_mean.binaryproto` - does this file exist? Has it been properly created, i.e. are the blob axes positive? See #6087 which is somewhat related.

Also, please do not post usage, installation, or modeling questions, or other requests for help to Issues. Use the [caffe-users list][1] instead. This helps developers maintain a clear, uncluttered, and efficient view of the state of Caffe. Please read the [guidelines for contributing][2] before submitting an issue or a pull request.

 [1]: https://groups.google.com/forum/#!forum/caffe-users
 [2]: https://github.com/BVLC/caffe/blob/master/CONTRIBUTING.md

**Comment by yyyjn** [comment]
请问这个问题解决了吗？？我也遇到了相同的错误。拜托

**Comment by murrayliu** [comment]
This is most likely caused by the mean file being malformed or non existent.


Andyyjn <notifications@github.com> 於 2020年6月6日 週六 下午4:46寫道：

**Comment by yyyjn** [comment]
那是要修改为.npy的格式吗还是？
